### PR TITLE
feat: improve performance of highlighter

### PR DESCRIPTION
**Pull request by uga-rosa**

### CODE REVIEW

File: lua/ccc/highlighter.lua [hunk 1]
```diff
@@ -38,107 +37,143 @@ function Highlighter:init()
...
+        api.nvim_buf_add_highlight(bufnr, self.picker_ns_id, hl_name, row, start - 1, end_)
+        init = end_ + 1
+    end
+end
+
+---@param bufnr integer
+function Highlighter:update_lsp(bufnr)
+    api.nvim_buf_clear_namespace(bufnr, self.lsp_ns_id, 0, -1)
+    local param = { textDocument = vim.lsp.util.make_text_document_params() }
+    vim.lsp.buf_request(bufnr, "textDocument/documentColor", param, function(err, colors)
```
Comment (hrsh7th) [inline lua/ccc/highlighter.lua:134]: I don't recommend to use `vim.lsp.buf_request`.

If multiple lsp servers were attached to the target buffer, this method will callback multiple times.

I recommend to use `vim.lsp.get_active_clients()` and you should check servers capabilities.

Comment (uga-rosa) [reply lua/ccc/highlighter.lua]: Is this correct what you are saying?

**Comment by uga-rosa** [comment]
@lopi-py How about now?

**Comment by lopi-py** [comment]
Yeah thats working properly now, the only thing that is a little annoying is that the first time I enter a buffer it doesn't highlight with lsp because it take a little to load

**Comment by uga-rosa** [comment]
You have to wait for the first LS initialization to highlight, of course. It would be possible to improve it after the second time, when the LS has already been initialized, but the work will be done tomorrow.

**Comment by lopi-py** [comment]
Sure, take your time!

**Pull request merged**